packages/react-instantsearch-dom/src/components/Stats.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { translatable } from 'react-instantsearch-core';
+import { createClassNames } from '../core/utils';
+
+const cx = createClassNames('Stats');
+
+export type StatsComponent = {
+  className?: string;
+  isSmartSorted: boolean;
+  nbHits: number;
+  nbSortedHits?: number;
+  processingTimeMS: number;
+  translate(
+    key: string,
+    n: number,
+    ms: number,
+    nSorted?: number,
+    isSmartSorted?: boolean
+  ): string;
+};
+
+const Stats: React.FC<StatsComponent> = ({
+  className = '',
+  isSmartSorted,
+  nbHits,
+  nbSortedHits,
+  processingTimeMS,
+  translate,
+}) => {
+  return (
+    <div className={classNames(cx(''), className)}>
+      <span className={cx('text')}>
+        {translate(
+          'stats',
+          nbHits,
+          processingTimeMS,
+          nbSortedHits,
+          isSmartSorted
+        )}
+      </span>
+    </div>
+  );
+};
+
+Stats.propTypes = {
+  className: PropTypes.string,
+  isSmartSorted: PropTypes.bool.isRequired,
+  nbHits: PropTypes.number.isRequired,
+  nbSortedHits: PropTypes.number,
+  processingTimeMS: PropTypes.number.isRequired,
+  translate: PropTypes.func.isRequired,
+};
+
+export default translatable({
+  stats: (
+    n: number,
+    ms: number,
+    nSorted?: number,
+    isSmartSorted?: boolean
+  ): string =>
+    isSmartSorted && nSorted !== undefined && n !== nSorted
```

Comment:
```suggestion
    isSmartSorted && n !== nSorted
```

is this part of the condition needed?

Reply:
As `nSorted` and `isSmartSorted` are optional, TypeScript will yet at line 64 if this condition is not here

Reply:
annoying, really isSmartSorted already guarantees that they're not undefined, so I'd use `!` on like 64

Reply:
Make sense! https://github.com/algolia/react-instantsearch/pull/3010/commits/c2e526bfababc9b42557c0f212a81e1a750b73a7